gpt4all-chat/qml/ChatView.qml

```diff
@@ -1694,18 +1694,22 @@ Rectangle {
                         imageHeight: 20
                         visible: chatModel.count && !currentChat.isServer && currentChat.isModelLoaded && !currentChat.responseInProgress
                         onClicked: {
-                            var index = Math.max(0, chatModel.count - 1);
-                            var listElement = chatModel.get(index);
+                            if (chatModel.count < 2)
+                                return
+                            var promptIndex = chatModel.count - 2
+                            var promptElement = chatModel.get(promptIndex)
+                            var responseIndex = chatModel.count - 1
+                            var responseElement = chatModel.get(responseIndex)
                             currentChat.regenerateResponse()
-                            if (chatModel.count) {
-                                if (listElement.name === "Response: ") {
-                                    chatModel.updateCurrentResponse(index, true);
-                                    chatModel.updateStopped(index, false);
-                                    chatModel.updateThumbsUpState(index, false);
-                                    chatModel.updateThumbsDownState(index, false);
-                                    chatModel.updateNewResponse(index, "");
-                                    currentChat.prompt(listElement.prompt)
-                                }
+                            if (responseElement.name === "Response: ") {
```

Comment:
Also, instead of two "if" statements like this, I think we should have another precondition check before regenerateResponse:
```js
if (promptElement.name != "Prompt: " || responseElement.name != "Response: ")
    return
```
It's easier to understand IMO if out of regenerateResponse(), updateCurrentResponse(), and currentChat.prompt(), either all happen or none happen.